### PR TITLE
Remove double newline at EOF of po files

### LINKED ISSUE #799: Two <BLANKLINE> in generated PO EOFs
As is documented, [`write_po` function renders two blank lines at the end of the file](https://github.com/python-babel/babel/blob/96acf8688c0e4b96788fbc35bb6d1640ef46a091/babel/messages/pofile.py#L497-L498). GNU gettext [documentation about PO files](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html) does not specify that this is mandatory.

I think that this makes no sense, but maybe something could explain why this has been the choice. It creates inconsistent output in diffs, uneeded extra space in disk, and so on... it is just a bad practice for text files.

### PR DESCRIPTION
**Pull request by CorentinDeBoisset**
Hello !

This is my first PR, so I'm not fully sure about everything. I used the unit tests to check for regressions, but are there other untested edge-cases we should worry about ?

Fixes #799

### CODE REVIEW

File: babel/messages/pofile.py [hunk 1]
```diff
@@ -665,10 +665,19 @@ def _format_message(message, prefix=''):
...
+        is_header = not message.id
+        if is_header and omit_header:
+            continue
+
+        if first_message:
+            first_message = False
+        else:
+            yield '\n'
+
+        if is_header:
```
Comment (akx) [inline babel/messages/pofile.py:680]: Would this become simpler (not sure if it would!) if `first_message` was instead something like `emitted_messages` (boolean or integer)? IOW, the `if not message.id: if omit_header: continue` check would be as is in the preimage, and then something like

```
   if emitted_messages:
       # Emit a separator between this and the previous message
       yield '\n'
   emitted_messages = True
```
?

**Comment by akx** [comment]
Thanks for the contribution! Please see the review comment; I'm not sure if it'd simplify things, but to me it looks like there's maybe more `if`s than necessary :)
